include/net/ieee802154_radio.h

```diff
@@ -157,15 +157,13 @@ struct ieee802154_radio_api {
 	u16_t (*get_subg_channel_count)(struct device *dev);
 #endif /* CONFIG_NET_L2_IEEE802154_SUB_GHZ */
 
-#ifdef CONFIG_NET_L2_OPENTHREAD
 	/** Run an energy detection scan.
-	 * Note: channel must be set prior to request this function.
-	 * duration parameter is in ms.
+	 *  Note: channel must be set prior to request this function.
+	 *  duration parameter is in ms.
 	 */
 	int (*ed_scan)(struct device *dev,
```

Comment:
Instead of adding dummy ed_scan API function to all the drivers, could we check whether the ed_scan pointer is set before calling it. This way we do not really need to add these dummy functions to all drivers. What do you think?

Reply:
Sure, sounds very reasonable, will fix.

Reply:
Removed stubs, added a check in the OT radio layer.